### PR TITLE
We've already added `ZLIB_LIBRARIES` to `LIBGIT2_LIBS` so don't also add the `z` library

### PR DESCRIPTION
**Pull request by dbrnz**
This resolves issue #5079.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -327,12 +327,6 @@ IF(NOT USE_BUNDLED_ZLIB)
 	IF(ZLIB_FOUND)
 		LIST(APPEND LIBGIT2_SYSTEM_INCLUDES ${ZLIB_INCLUDE_DIRS})
 		LIST(APPEND LIBGIT2_LIBS ${ZLIB_LIBRARIES})
-		IF(APPLE OR CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
-			LIST(APPEND LIBGIT2_LIBS "z")
-			LIST(APPEND LIBGIT2_PC_LIBS "-lz")
```
Comment (pks-t) [inline src/CMakeLists.txt]: I agree that the first line is definitely bogus, we don't want to append twice to `LIBGIT2_LIBS`. But appending to `LIBGIT2_PC_LIBS` is in fact required. Doing so will not cause us to link twice, but the `PC_LIBS` variable will cause us to append this lib to the generated pkgconfig file.

Comment (dbrnz) [reply src/CMakeLists.txt]: Agree -- updates committed.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -327,12 +327,11 @@ IF(NOT USE_BUNDLED_ZLIB)
...
-		ELSE()
-			LIST(APPEND LIBGIT2_PC_REQUIRES "zlib")
-		ENDIF()
+        	IF(APPLE OR CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
+            		LIST(APPEND LIBGIT2_PC_LIBS "-lz")
+        	ELSE()
+            		LIST(APPEND LIBGIT2_PC_REQUIRES "zlib")
+        	ENDIF()
```
Comment (pks-t) [inline src/CMakeLists.txt]: Sorry to be a nag, but there's some leading spaces now in front of the tabs. If you convert that to all-tabs then we're ready to go!

Comment (dbrnz) [reply src/CMakeLists.txt]: Tabs versus spaces can be a pain! Hopefully all is now good.

**Pull request merged**